provider/pkg/schema/metadata.go

```diff
@@ -12,11 +12,17 @@ type CloudAPIMetadata struct {
 
 // CloudAPIResource contains metadata for a single AWS Resource.
 type CloudAPIResource struct {
-	CfType     string                          `json:"cf"`
-	Inputs     map[string]pschema.PropertySpec `json:"inputs"`
-	Outputs    map[string]pschema.PropertySpec `json:"outputs"`
-	Required   []string                        `json:"required,omitempty"`
-	CreateOnly []string                        `json:"createOnly,omitempty"`
+	CfType          string                          `json:"cf"`
+	Inputs          map[string]pschema.PropertySpec `json:"inputs"`
+	Outputs         map[string]pschema.PropertySpec `json:"outputs"`
+	AutoNamingSpecs map[string]AutoNamingSpec       `json:"autoNamingSpecs,omitempty"`
```

Comment:
Are there any cases when we auto-name more than one property?

Reply:
No, not with the current heuristic. I will replace with just a reference to `AutoNamingSpec`.